src/nupic/torch/modules/sparse_weights.py

```diff
@@ -134,6 +138,18 @@ class SparseWeights(SparseWeightsBase):
     def __init__(self, module, weight_sparsity=None, sparsity=None,
                  allow_extremes=False):
         assert len(module.weight.shape) == 2, "Should resemble a nn.Linear"
+
+        # Add checks to ensure there is no incompatibility
+        if sparsity is None:
```

Comment:
Maybe an easier check would be to make sure only one of these parameters are passed.

Reply:
```
 assert not(sparsity and weight_sparsity)
```

Reply:
Ok, I can change to that